### PR TITLE
fix(facebook): on mobile, facebook can't detect user's account

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # mf-251

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 1]
```diff
@@ -22,18 +22,24 @@ export const searchAvatarSelector: () => LiveSelector<HTMLImageElement, true> =
 export const searchNickNameSelector: () => LiveSelector<HTMLHeadingElement, true> = () =>
     querySelector<HTMLHeadingElement>('span[dir="auto"] div h1')
 
-export const searchNickNameSelectorOnMobile: () => LiveSelector<E, true> = () => querySelector<E>('#cover-name-root h3')
-
+// export const searchNickNameSelectorOnMobile: () => LiveSelector<E, true> = () => querySelector<E>('#cover-name-root h3')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts:25]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 2]
```diff
@@ -22,18 +22,24 @@ export const searchAvatarSelector: () => LiveSelector<HTMLImageElement, true> =
...
+export const searchNickNameSelectorOnMobile: () => LiveSelector<E, true> = () => querySelector<E>('#screen-root h1')
+export const searchNickNameSelectorOnMobileBig: () => LiveSelector<E, true> = () =>
+    querySelector<E>('#rootcontainer h3')
 export const bioDescriptionSelector = () =>
     querySelector('span[dir="auto"] div h1').closest(7).querySelector<HTMLSpanElement>('span[dir="auto"] > span')
 
-export const bioDescriptionSelectorOnMobile: () => LiveSelector<HTMLDivElement, true> = () => querySelector('#bio div')
+// export const bioDescriptionSelectorOnMobile: () => LiveSelector<HTMLDivElement, true> = () => querySelector('#bio div')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts:32]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 3]
```diff
@@ -47,9 +53,11 @@ export const searchFacebookAvatarMobileListSelector = () => querySelector('#nuxC
 export const searchFacebookAvatarSelector = () =>
     querySelector('[role="main"] [role="button"] svg[role="img"],[role="main"] [role="link"] svg[role="img"]')
 
-export const searchFacebookAvatarOnMobileSelector = () =>
-    querySelector('[data-sigil="timeline-cover"] i[aria-label$="picture"]')
+// export const searchFacebookAvatarOnMobileSelector = () =>
+//    querySelector('[data-sigil="timeline-cover"] i[aria-label$="picture"]')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts:57]:
```suggestion
```

File: packages/mask/src/social-network-adaptor/facebook.com/utils/user.ts [hunk 4]
```diff
@@ -16,7 +17,9 @@ import { isMobileFacebook } from './isMobile'
 import { bioDescription, personalHomepage } from '../../../settings/settings'
 
 export const getNickName = () => {
-    const node = isMobileFacebook ? searchNickNameSelectorOnMobile().evaluate() : searchNickNameSelector().evaluate()
+    const node = isMobileFacebook
+        ? searchNickNameSelectorOnMobile().evaluate() || searchNickNameSelectorOnMobileBig().evaluate()
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/user.ts:21]: Can we merge `searchNickNameSelectorOnMobileBig` into `searchNickNameSelectorOnMobile`?

**Pull request closed without merging**